src/Microsoft.AspNet.Http.Extensions/HeaderDictionaryExtensions.cs

```diff
@@ -36,7 +36,7 @@ public static void AppendCommaSeparatedValues(this IHeaderDictionary headers, st
         /// <returns>the associated values from the collection separated into individual values, or StringValues.Empty if the key is not present.</returns>
         public static string[] GetCommaSeparatedValues(this IHeaderDictionary headers, string key)
         {
-            return ParsingHelpers.GetHeaderSplit(headers, key);
+            return ParsingHelpers.GetHeaderSplit(headers, key).ToArray();
```

Comment:
Much shame; corner cases and too many changes for a single PR - lesson learnt. Seems to have engaged people though :D